radix-engine/src/blueprints/pool/single_resource_pool/substates.rs

```diff
@@ -0,0 +1,24 @@
+use native_sdk::resource::*;
+use radix_engine_common::data::scrypto::model::*;
+use radix_engine_common::prelude::*;
+use radix_engine_common::*;
+use radix_engine_interface::blueprints::resource::*;
+
+#[derive(Debug, PartialEq, Eq, ScryptoSbor, Clone)]
+pub struct SingleResourcePoolSubstate {
+    /// The vault of the resources of the pool.
+    pub vault: Own,
+
+    /// The address of the pool unit resource that the pool works with.
+    pub pool_unit_resource: ResourceAddress,
```

Comment:
This could just be a `ResourceManager`.

Reply:
Nice! Will update it. I guess `Own` could also just be `Vault`?

Reply:
Yup, I believe so.

Reply:
Nice! I went ahead and did that. `Vault` does not implement `Clone` so all substate types that need to be Cloneable implement a custom clone.